### PR TITLE
Update correctly ant template to gradle

### PR DESCRIPTION
**Pull request by Fatme**

### CODE REVIEW

File: lib/services/android-project-service.ts [hunk 1]
```diff
@@ -153,11 +153,21 @@ class AndroidProjectService extends projectServiceBaseLib.PlatformProjectService
 	}
 
 	public canUpdatePlatform(currentVersion: string, newVersion: string): IFuture<boolean> {
-		return Future.fromResult<boolean>(true);
+		return (() => {
```
Comment (teobugslayer) [inline lib/services/android-project-service.ts]: perhaps `return Future.fromResult(true)`?

File: lib/services/android-project-service.ts [hunk 2]
```diff
@@ -153,11 +153,21 @@ class AndroidProjectService extends projectServiceBaseLib.PlatformProjectService
...
+			return true;
+		}).future<boolean>()();
 	}
 
-	public updatePlatform(currentVersion: string, newVersion: string): IFuture<void> {
-		return Future.fromResult();
+	public updatePlatform(currentVersion: string, newVersion: string, canUpdate: boolean): IFuture<boolean> {
```
Comment (teobugslayer) [inline lib/services/android-project-service.ts]: I'd really like to see documented what return values `true` and `false` signify.

File: lib/services/ios-project-service.ts [hunk 3]
```diff
@@ -240,24 +241,33 @@ export class IOSProjectService extends projectServiceBaseLib.PlatformProjectServ
...
-			this.$logger.info("Copied from %s at %s.", cachedPackagePath, this.platformData.projectRoot);
-
-			let pbxprojFilePath = path.join(this.platformData.projectRoot, this.$projectData.projectName + IOSProjectService.XCODE_PROJECT_EXT_NAME, "project.pbxproj");
-			this.replaceFileContent(pbxprojFilePath).wait();
-		}).future<void>()();
+			if(!canUpdate) {
+				let isUpdateConfirmed = this.$prompter.confirm(`We need to override xcodeproj file. The old one will be saved at ${this.$options.profileDir}. Are you sure?`, () => true).wait();
+				if(isUpdateConfirmed) {
+					// Copy old file to options["profile-dir"]
+					let sourceFile = path.join(this.platformData.projectRoot, util.format("%s.xcodeproj", this.$projectData.projectName));
```
Comment (teobugslayer) [inline lib/services/ios-project-service.ts]: You can use templated strings here, but not a merge stopper.

File: lib/services/ios-project-service.ts [hunk 4]
```diff
@@ -240,24 +241,33 @@ export class IOSProjectService extends projectServiceBaseLib.PlatformProjectServ
...
+					let destinationFile = path.join(this.$options.profileDir, "xcodeproj");
+					this.$fs.deleteDirectory(destinationFile).wait();
+					shell.cp("-R", path.join(sourceFile, "*"), destinationFile);
+					this.$logger.info("Backup file %s at location %s", sourceFile, destinationFile);
+					this.$fs.deleteDirectory(path.join(this.platformData.projectRoot, util.format("%s.xcodeproj", this.$projectData.projectName))).wait();
```
Comment (teobugslayer) [inline lib/services/ios-project-service.ts]: Isn't this `deleteDirectory(sourceFile)`?

File: lib/services/ios-project-service.ts [hunk 5]
```diff
@@ -240,24 +241,33 @@ export class IOSProjectService extends projectServiceBaseLib.PlatformProjectServ
...
+
+					// Copy xcodeProject file
+					let cachedPackagePath = path.join(this.$npmInstallationManager.getCachedPackagePath(this.platformData.frameworkPackageName, newVersion), constants.PROJECT_FRAMEWORK_FOLDER_NAME, util.format("%s.xcodeproj", IOSProjectService.IOS_PROJECT_NAME_PLACEHOLDER));
```
Comment (teobugslayer) [inline lib/services/ios-project-service.ts]: Again, you can use template string, but not a merge stopper.

File: lib/services/ios-project-service.ts [hunk 6]
```diff
@@ -240,24 +241,33 @@ export class IOSProjectService extends projectServiceBaseLib.PlatformProjectServ
...
+					shell.cp("-R", path.join(cachedPackagePath, "*"), path.join(this.platformData.projectRoot, util.format("%s.xcodeproj", this.$projectData.projectName)));
```
Comment (teobugslayer) [inline lib/services/ios-project-service.ts]: again, can you use `sourceFile` here?

File: lib/services/android-project-service.ts [hunk 7]
```diff
@@ -152,11 +153,19 @@ class AndroidProjectService extends projectServiceBaseLib.PlatformProjectService
...
-		return Future.fromResult<boolean>(true);
+		return Future.fromResult(true);
 	}
 
-	public updatePlatform(currentVersion: string, newVersion: string): IFuture<void> {
-		return Future.fromResult();
+	public updatePlatform(currentVersion: string, newVersion: string, canUpdate: boolean): IFuture<boolean> {
+		return (() => {
+			if(semver.eq(newVersion, AndroidProjectService.MIN_RUNTIME_VERSION_WITH_GRADLE)) {
+				this.$commandsService.tryExecuteCommand(process.argv[2], ["remove"].concat(process.argv.slice(4))).wait();
```
Comment (teobugslayer) [inline lib/services/android-project-service.ts]: Imho, this is a highly suspicious code - our project service depends on the UI code, which means that a lower-layered code depends on the higher level one. Can we approach this in another way?

File: lib/services/android-project-service.ts (hunk 2 above)
Comment (Fatme) [inline lib/services/android-project-service.ts]: Done, you can check the last commit :)

File: lib/services/android-project-service.ts (hunk 7 above)
Comment (Fatme) [inline lib/services/android-project-service.ts]: The another approach is to use `childProcess.spawn` but I'm not sure if this is a better choice.  I'm opened to any suggestions :)

**Comment by teobugslayer** [comment]
Squash and go, :+1:

**Pull request merged**